src/main/java/de/tum/in/www1/artemis/service/QuizExerciseImportService.java

```diff
@@ -38,17 +44,18 @@ public QuizExerciseImportService(QuizExerciseService quizExerciseService, FileSe
      * Imports a quiz exercise creating a new entity, copying all basic values and saving it in the database.
      * All basic include everything except Student-, Tutor participations, and student questions. <br>
      * This method calls {@link #copyQuizExerciseBasis(QuizExercise)} to set up the basis of the exercise and
-     * {@link #copyQuizQuestions(QuizExercise, QuizExercise)} for a hard copy of the questions.
+     * {@link #copyQuizQuestions(QuizExercise, QuizExercise, List)} for a hard copy of the questions.
      *
      * @param templateExercise The template exercise which should get imported
      * @param importedExercise The new exercise already containing values which should not get copied, i.e. overwritten
+     * @param providedFiles    The files that were provided by the user
      * @return The newly created exercise
      */
     @NotNull
-    public QuizExercise importQuizExercise(final QuizExercise templateExercise, QuizExercise importedExercise) {
+    public QuizExercise importQuizExercise(final QuizExercise templateExercise, QuizExercise importedExercise, @Nonnull List<MultipartFile> providedFiles) throws IOException {
```

Comment:
We usually use `javax.validation.constraints.NotNull` instead of `javax.validation.Nonnull`.

Reply:
God, why are there so many nonnull/notnull annotations?